build.sbt

```diff
@@ -1,111 +1,69 @@
-  "com.typesafe.akka"          %% "akka-stream-testkit"         % akkaVersion % Test,
-  "com.danielasfregola"        %% "random-data-generator"       % "2.7" % Test,
-  "com.47deg"                  %% "scalacheck-toolbox-datetime" % "0.2.5" % Test,
-  "org.mockito"                % "mockito-all"                  % "1.10.19" % Test,
-  "org.zalando"                %% "grafter"                     % "2.6.1" % Test,
-  "io.github.embeddedkafka"    %% "embedded-kafka"              % "2.6.0" % Test
-)
+Test / testOptions += Tests.Argument(TestFrameworks.ScalaTest, "-oF")
 
 val commonSettings = Seq(
   organization := "com.sky",
-  scalaVersion := "2.12.10",
-  libraryDependencies += "com.sksamuel.avro4s" %% "avro4s-core" % "2.0.2"
+  scalaVersion := "2.13.8"
+)
+
+val compilerSettings = Seq(
+  // Compiler option not provided by sbt-tpolecat
+  scalacOptions += "-Ymacro-annotations",
+  tpolecatScalacOptions ~= { opts =>
+    opts.filterNot(Set(ScalacOptions.warnValueDiscard))
+  }
 )
 
 lazy val dockerSettings = Seq(
-  packageName in Docker := "kafka-message-scheduler",
-  dockerBaseImage := "alpine:3.13.2",
-  dockerRepository := Some("skyuk"),
-  dockerLabels := Map("maintainer" -> "Sky"),
-  dockerUpdateLatest := true,
-  dockerCommands ++= Seq(
-    Cmd("USER", "root"),
-    Cmd("RUN", "apk add --no-cache bash eudev openjdk11-jre")
-  )
+  Docker / packageName := "kafka-message-scheduler",
+  dockerBaseImage      := "gcr.io/distroless/java17-debian11:nonroot",
+  dockerRepository     := Some("skyuk"),
+  dockerLabels         := Map("maintainer" -> "Sky"),
+  dockerUpdateLatest   := true
 )
 
 val buildInfoSettings = Seq(
-  buildInfoKeys := Seq[BuildInfoKey](name, version, scalaVersion, sbtVersion),
+  buildInfoKeys    := Seq[BuildInfoKey](name, version, scalaVersion, sbtVersion),
   buildInfoPackage := "com.sky"
 )
 
 lazy val scheduler = (project in file("scheduler"))
   .enablePlugins(BuildInfoPlugin, JavaAppPackaging, UniversalDeployPlugin, JavaAgent, DockerPlugin)
   .settings(commonSettings)
+  .settings(compilerSettings)
   .settings(
-    libraryDependencies ++= dependencies,
-    dependencyOverrides ++= Seq("org.apache.kafka" % "kafka-clients" % kafkaVersion,
-                                "org.scalacheck"   %% "scalacheck"   % "1.13.5"),
-    resolvers ++= Seq(
-      "jitpack" at "https://jitpack.io",
-      Resolver.bintrayRepo("cakesolutions", "maven")
-    ),
-    addCompilerPlugin("org.scalamacros" % "paradise"        % "2.1.1" cross CrossVersion.full),
-    addCompilerPlugin("org.spire-math"  %% "kind-projector" % "0.9.10"),
-    scalacOptions ++= Seq(
-      "-language:implicitConversions",
-      "-language:postfixOps",
-      "-Xfatal-warnings",
-      "-Ywarn-dead-code",
-      "-Ywarn-unused",
-      "-deprecation",
-      "-Ypartial-unification",
-      "-encoding",
-      "utf-8"
+    libraryDependencies ++= Dependencies.all,
+    dependencyOverrides ++= Seq(
+      Dependencies.Kafka.kafkaClients,
```

Comment:
Can we include kafkaClients as part of our `libraryDependencies` as well because we directly use it in the scheduler code.